### DISCUSSION TITLE
(Synced Lyrics) Does track length affect the insertion of synchronized Lyrics?

### DISCUSSION THREAD

**Discussion by Dzheremi2** [pear-devs/pear-desktop#2333] [Q&A] [answered]
Does anyone know, if track length affects the insertion of synchronized Lyrics?

**Comment by ArjixWasTaken** [comment]
yes it does

**Reply by ArjixWasTaken** [reply, marked as answer]
The code picks the lyrics result that is closest to the track's duration.
https://github.com/th-ch/youtube-music/blob/master/src/plugins/synced-lyrics/renderer/lyrics/fetch.ts#L139-L152

If the difference from the closest result to the track's duration is more than 15 seconds it doesn't insert lyrics
https://github.com/th-ch/youtube-music/blob/master/src/plugins/synced-lyrics/renderer/lyrics/fetch.ts#L156

if the difference is more than 5, but less than 15, it shows a warning that the timings may not be accurate
https://github.com/th-ch/youtube-music/blob/master/src/plugins/synced-lyrics/renderer/lyrics/fetch.ts#L157

**Reply by ArjixWasTaken** [reply]
I should also mention that the results are filtered based on how similar the artist's name is to the artist of the track.
https://github.com/th-ch/youtube-music/blob/master/src/plugins/synced-lyrics/renderer/lyrics/fetch.ts#L126-L137

If anyone has a better algorithm to propose, please make an issue

**Reply by Lockszmith-GH** [reply]
Don't know about 'better', but I really wish there would be a manual override, where I can enter a search phrase - this can be especially useful with Videos, and with non-english language songs.

**Reply by ArjixWasTaken** [reply]
Manually selecting the lyrics sounds great, but I would consider that as a different feature.

I do have plans for that when I am done introducing multiple lyric sources.

(I've been inactive due to uni exams)
